src/main/java/org/kuali/kfs/krad/service/impl/RemoteModuleServiceBase.java

```diff
@@ -0,0 +1,653 @@
+        this.moduleConfiguration = moduleConfiguration;
+    }
+
+    /**
+     * @see ModuleService#isExternalizable(java.lang.Class)
+     */
+    @Override
+    public boolean isExternalizable(Class boClazz) {
+        if (boClazz == null) {
+            return false;
+        }
+        return ExternalizableBusinessObject.class.isAssignableFrom(boClazz);
+    }
+
+    public <T extends ExternalizableBusinessObject> T createNewObjectFromExternalizableClass(Class<T> boClass) {
+        try {
+            return (T) getExternalizableBusinessObjectImplementation(boClass).newInstance();
+        } catch (Exception e) {
+            throw new RuntimeException("Unable to create externalizable business object class", e);
+        }
+    }
+
+    public DataObjectRelationship getBusinessObjectRelationship(Class boClass, String attributeName,
+                                                                String attributePrefix) {
+        return null;
+    }
+
+
+    /**
+     * @return the kualiModuleService
+     */
+    public KualiModuleService getKualiModuleService() {
+        return this.kualiModuleService;
+    }
+
+    /**
+     * @param kualiModuleService the kualiModuleService to set
+     */
+    public void setKualiModuleService(KualiModuleService kualiModuleService) {
+        this.kualiModuleService = kualiModuleService;
+    }
+
+    protected ConfigurationService getKualiConfigurationService() {
+        if (this.kualiConfigurationService == null) {
+            this.kualiConfigurationService = KRADServiceLocator.getKualiConfigurationService();
+        }
+
+        return this.kualiConfigurationService;
+    }
+
+    public void setKualiConfigurationService(ConfigurationService kualiConfigurationService) {
+        this.kualiConfigurationService = kualiConfigurationService;
+    }
+
+    /**
+     * @see org.springframework.context.ApplicationContextAware#setApplicationContext(org.springframework.context.ApplicationContext)
+     */
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+        this.applicationContext = applicationContext;
+    }
+
+    /**
+     * This overridden method ...
+     *
+     * @see ModuleService#listAlternatePrimaryKeyFieldNames(java.lang.Class)
+     */
+    @Override
+    public List<List<String>> listAlternatePrimaryKeyFieldNames(Class businessObjectInterfaceClass) {
+        return null;
+    }
+
+    /**
+     * This method determines whether or not this module is currently locked
+     *
+     * @see ModuleService#isLocked()
+     */
+    @Override
+    public boolean isLocked() {
+        LOG.debug("isLocked, entering Cornell customization");
+        boolean shouldLock = false;
+        ModuleConfiguration configuration = this.getModuleConfiguration();
+        if (configuration != null) {
+            String namespaceCode = configuration.getNamespaceCode();
```

Comment:
Since this method has grown in length significantly it might make sense to extract the code that checks if the module is locked into a separate method: isModuleLocked. Then this method could return something like: 
isModuleLocked && isUserLocked.